extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/AuthenticationProviderService.java

```diff
@@ -155,9 +157,10 @@ public AuthenticatedUser authenticateUser(Credentials credentials)
         // This is a response to a previous challenge, authenticate with that.
         else {
             try {
+                byte[] stateBytes = javax.xml.bind.DatatypeConverter.parseHexBinary(request.getParameter(RadiusStateField.PARAMETER_NAME));
```

Comment:
Might not be a bad idea to handle the `IllegalArgumentException` case for `parseHexBinary()` either, so that the backend generally behaves in a robust manner regardless of what's thrown at it (safely catches and handles data which does not match our assumptions rather than exploding).

Reply:
At least I was consistent.

Reply:
Very true.

Reply:
Okay, removed the extra verbosity, handled the case where state field is null, and the `IllegalArgumentException`.